mne_connectivity/spectral/epochs_multivariate.py

```diff
@@ -287,31 +286,71 @@ def _compute_con_daughter(
-            ).transpose(1, 0, 2, 3)
-
-        if not np.isreal(T).all() or not np.isfinite(T).all():
+        try:
+            return self._invsqrtm(C_r, n_seeds)
+        except np.linalg.LinAlgError as error:
             raise RuntimeError(
-                "the transformation matrix of the data must be real-valued "
-                "and contain no NaN or infinity values; check that you are "
-                "using full rank data or specify an appropriate rank for the "
-                "seeds and targets that is less than or equal to their ranks"
-            )
-
-        return np.real(T)  # make T real if check passes
+                "the transformation matrix of the data could not be computed "
+                "from the cross-spectral density; check that you are using "
+                "full rank data or specify an appropriate rank for the seeds "
+                "and targets that is less than or equal to their ranks"
+            ) from error
+
+    def _invsqrtm(self, C_r, n_seeds):
+        """Compute inverse sqrt of CSD over frequencies and times.
+
+        Parameters
+        ----------
+        C_r : np.ndarray, shape=(n_freqs, n_times, n_channels, n_channels)
+            Real part of the CSD. Expected to be symmetric and non-singular.
+        n_seeds : int
+            Number of seed channels for the connection.
+
+        Returns
+        -------
+        T : np.ndarray, shape=(n_freqs, n_times, n_channels, n_channels)
+            Inverse square root of the real-valued CSD. Name comes from Ewald
+            et al. (2012).
+
+        Notes
+        -----
+        This approach is a workaround for computing the inverse square root of
+        an ND array. SciPy has dedicated functions for this purpose, e.g.
+        `sp.linalg.fractional_matrix_power(A, -0.5)` or `sp.linalg.inv(
+        sp.linalg.sqrtm(A))`, however these only work with 2D arrays, meaning
+        frequencies and times must be looped over which is very slow. There are
+        no equivalent functions in NumPy for working with ND arrays (as of
+        v1.26).
+
+        The data array is expected to be symmetric and non-singular, otherwise
+        a LinAlgError is raised.
+
+        See Eq. 3 of Ewald et al. (2012). NeuroImage. DOI:
+        10.1016/j.neuroimage.2011.11.084.
+        """
+        T = np.zeros_like(C_r, dtype=np.float64)
+
+        # seeds
+        eigvals, eigvects = np.linalg.eigh(C_r[:, :, :n_seeds, :n_seeds])
+        if (eigvals == 0).any():  # sign of non-full rank data
+            raise np.linalg.LinAlgError()
+        T[:, :, :n_seeds, :n_seeds] = (
+            eigvects * np.expand_dims(1.0 / np.sqrt(eigvals), (2))
+        ) @ eigvects.transpose(0, 1, 3, 2)
+
+        # targets
+        eigvals, eigvects = np.linalg.eigh(C_r[:, :, n_seeds:, n_seeds:])
+        if (eigvals == 0).any():  # sign of non-full rank data
+            raise np.linalg.LinAlgError()
```

Comment:
```suggestion
        n_zero = (eigvals == 0).sum()
        if n_zero:  # sign of non-full rank data
            raise np.linalg.LinAlgError(
                "Cannot compute inverse square root of rank-deficient matrix "
                f"with {n_zero}/{len(eigvals)} zero eigenvalue(s)"
            )
```

Reply:
Sure, added for both the seed and target eigvals.